examples/plot_04_competing_risks.py

```diff
@@ -0,0 +1,167 @@
+"""
+==============================
+Competing Risks Example
+==============================
+
+In this notebook, we showcase how to use the `SurvivalBoost` class to model
+competing risks data. We generate a synthetic dataset with three competing
+events and fit a model to predict the cumulative incidence functions and the
+survival function.
+"""
+
+# %%
+# Usage
+# =====
+#
+# Generating synthetic data
+# -------------------------
+#
+# We begin by generating a linear, synthetic dataset with three competing events.
+
+import numpy as np
+from hazardous.data import make_synthetic_competing_weibull
+
+np.random.seed(0)
+
+n_events = 3
+X, y = make_synthetic_competing_weibull(
+    n_events=n_events, n_samples=10_000, return_X_y=True
+)
+
+# %%
+# Next, we display the distribution of our target. Event 0 corresponds to censoring.
+import seaborn as sns
+from matplotlib import pyplot as plt
+
+
+sns.histplot(
+    y,
+    x="duration",
+    hue="event",
+    multiple="stack",
+    palette="colorblind",
+)
+plt.show()
+
+# %%
+from sklearn.model_selection import train_test_split
+from hazardous import SurvivalBoost
+
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+survival_boost = SurvivalBoost(n_iter=10, show_progressbar=False).fit(X_train, y_train)
+survival_boost
+
+# %%
+
+incidence_curves = survival_boost.predict_cumulative_incidence(
+    X_test,
+    times=None,
+)
+
+survival_curves = incidence_curves[:, 0]  # survival function S(t)
+
+# %%
+# Let's plot the estimated survival function and the incidence functions for
+# some patients.We will also plot a symbol at the time of death or censoring.
+# The symbol will be a skull and crossbones for the event that occurred,
+# a cross for meaning that an other event of interest has happened, and a
+# question mark for the censoring event.
+
+import matplotlib.pyplot as plt
+
+fig, ax = plt.subplots(nrows=1, ncols=4, figsize=(15, 4))
+sns.despine()
+patient_ids_to_plot = [0, 1, 2, 3]
+
+for event_id in range(n_events + 1):
```

Comment:
These plots are very helpful. I think we could include them in a function like `plot_survival_curves` in a `_plot.py` file so that they're easy to see. WDYT? 

It's ok to do it in a subsequent PR, though.